synapse/http/federation/matrix_federation_agent.py

```diff
@@ -211,13 +231,17 @@ def __init__(
             srv_resolver = SrvResolver()
 
         self._srv_resolver = srv_resolver
+        self.https_proxy_endpoint = proxyagent.http_proxy_endpoint(
+            https_proxy, proxy_reactor
+        )
```

Comment:
Shouldn't we not create this endpoint if `https_proxy` is not given?

Reply:
`http_proxy_endpoint(...)` returns `None` when you give it `None` as an input, so I think this is fine?